src/components/SplitButton/SplitButton.jsx

```diff
@@ -0,0 +1,215 @@
+				/**
+				 * Disables selection of the `Button`.
+				 */
+				isDisabled: bool,
+				/**
+				 * Called when the user clicks the `Button`.
+				 *
+				 * Signature: `({ event, props }) => {}`
+				 */
+				onClick: func,
+			},
+		}),
+	},
+
+	reducers,
+
+	propTypes: {
+		/**
+		 * Object of DropMenu props which are passed through to the underlying
+		 * DropMenu component.
+		 */
+		DropMenu: shape(DropMenu.propTypes),
+
+		/**
+		 * All children should be `ButtonGroup.Button`s and they support the same
+		 * props as `Button`s.
+		 */
+		children: any,
+
+		/**
+		 * Appended to the component-specific class names set on the root
+		 * element. Value is run through the `classnames` library.
+		 */
+		className: any,
+
+		/**
+		 * Sets the direction the flyout menu will render relative to the
+		 * SplitButton.
+		 */
+		direction: oneOf([
+			'up',
+			'down',
+		]),
+
+		/**
+		 * Style variations of the SplitButton.
+		 */
+		kind: oneOf([
+			'primary',
+			'success',
+			'warning',
+			'danger',
+			'info',
+		]),
+
+		/**
+		 * Size variations of the SplitButton.
+		 */
+		size: oneOf([
+			'short',
+			'small',
+			'large',
+		]),
+
+		/**
+		 * Form element type variations of SplitButton. Defaults to 'button' to avoid
+		 * being triggered by 'Enter' anywhere on the page. Passed through to DOM
+		 * Button.
+		 */
+		type: string,
+	},
+
+	getDefaultProps() {
+		return {
+			direction: 'down',
+			type: 'button',
+			DropMenu: DropMenu.getDefaultProps(),
+		};
+	},
+
+	// Handles clicks on the Primary Button
+	handleClick({event}) {
+		const clickedButtonProps = _.get(getFirst(this.props, SplitButton.Button), 'props');
+
+		// Stop propagation to prevent this `Click` from expanding the `DropMenu`
+		event.stopPropagation();
+		this.handleButtonClick(clickedButtonProps, event);
+	},
+
+	// Handles clicks in the DropMenu
+	handleSelect(optionIndex, {event}) {
+		const buttonChildProps = _.map(filterTypes(this.props.children, SplitButton.Button), 'props');
+		this.handleButtonClick(buttonChildProps[optionIndex + 1], event);
+	},
+
+	handleButtonClick(buttonProps, event) {
+		const {
+			DropMenu: {
+				onCollapse,
+			},
+		} = this.props;
+
+		onCollapse();
+
+		if (_.has(buttonProps, 'onClick')) {
+			buttonProps.onClick({event, props: buttonProps});
+		}
+	},
+
+	render() {
+		const {
+			className,
+			kind,
+			direction,
+			type,
+			size,
+			DropMenu: dropMenuProps,
+			...passThroughs
+		} = this.props;
+
+		const {
+			isExpanded,
+		} = dropMenuProps;
+
+		const [primaryButtonProps, ...buttonChildProps] = _.map(filterTypes(this.props.children, SplitButton.Button), 'props');
+
+		return (
+			<DropMenu
+				{...dropMenuProps}
+				{...omitProps(passThroughs, SplitButton)}
+				direction={direction}
+				className={cx('&', className)}
+				onSelect={this.handleSelect}
+			>
+				<DropMenu.Control>
+					<ButtonGroup>
+						<Button
+							{...primaryButtonProps}
+							className={cx('&-Primary-Button')}
```

Comment:
i think we should only capitalize parts of the className when it refers to a component type, `Button` makes sense but `primary` should be lowercase. Also i think the type should be front-loaded.